packages/react/src/components/Popover/index.js

```diff
@@ -23,20 +32,140 @@ const Popover = React.forwardRef(function Popover(props, ref) {
     ...rest
   } = props;
   const prefix = usePrefix();
+  const floating = useRef();
+  const popover = useRef();
+  const ref = useMergedRefs([forwardRef, popover]);
+  const [autoAligned, setAutoAligned] = useState(false);
+  const [autoAlignment, setAutoAlignment] = useState(align);
   const className = cx({
     [`${prefix}--popover-container`]: true,
     [`${prefix}--popover--caret`]: caret,
     [`${prefix}--popover--drop-shadow`]: dropShadow,
     [`${prefix}--popover--high-contrast`]: highContrast,
     [`${prefix}--popover--open`]: open,
-    [`${prefix}--popover--${align}`]: true,
+    [`${prefix}--popover--${autoAlignment}`]: autoAligned,
+    [`${prefix}--popover--${align}`]: !autoAligned,
     [customClassName]: !!customClassName,
   });
 
+  useIsomorphicEffect(() => {
+    if (!autoAlign) {
+      return;
+    }
+
+    if (!floating.current) {
+      return;
+    }
+
+    if (autoAligned === true) {
+      return;
+    }
+
+    const rect = floating.current.getBoundingClientRect();
+
+    const conditions = {
+      left: rect.x < -16,
+      top: rect.y < -16,
+      right: rect.x + (rect.width - 16) > document.documentElement.clientWidth,
+      bottom:
+        rect.y + (rect.height - 16) > document.documentElement.clientHeight,
+    };
```

Comment:
just wanna make sure I'm understanding correctly. this is comparing the popover measurements to the page margins? (-16?)

Reply:
wondering if you could maybe do a one or two liner comment summarizing what `conditions` is for, for maintainers?